v2/internal/reconcilers/generic/register.go

```diff
@@ -148,6 +156,8 @@ func register(
 		WithOptions(options.Options)
 	builder.Named(info.Name)
 
+	builder = builder.Watches(&corev1.Namespace{}, handler.EnqueueRequestsFromMapFunc(registerNamespaceWatcher(options, kubeClient, info.Obj, gvk)), ctrlbuilder.WithPredicates(predicate.AnnotationChangedPredicate{}))
```

Comment:
Please change this to use internal/reconcilers/predicates.go `ARMReconcilerAnnotationChangedPredicate`

That will filter the events to only the annotation we care about.